code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -1570,48 +1586,56 @@ func (srh *StorageRestHandler) getAllocationMinLock(w http.ResponseWriter, r *ht
 		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
 		return
 	}
-	blobbers, err := getBlobbersForRequest(req, edb, balances, common2.Pagination{})
+
+	var ssc = StorageSmartContract{SmartContract: sci.NewSC(ADDRESS)}
+	ssc.setSC(ssc.SmartContract, &ccsc.BCContext{})
+	conf, err := getConfig(balances)
 	if err != nil {
-		common.Respond(w, r, "", common.NewErrInternal("error selecting blobbers", err.Error()))
+		common.Respond(w, r, nil, common.NewErrInternal(err.Error()))
 		return
 	}
-	sa := req.storageAllocation()
-	var gbSize = sizeInGB(sa.bSize())
-	var minLockDemand currency.Coin
-
-	ids := append(req.Blobbers, blobbers...)
-	uniqueMap := make(map[string]struct{})
-	for _, id := range ids {
-		uniqueMap[id] = struct{}{}
+	var request newAllocationRequest
+	if err = request.decode([]byte(allocData)); err != nil {
+		common.Respond(w, r, nil, common.NewErrBadRequest(err.Error()))
+		return
 	}
-	unique := make([]string, 0, len(ids))
-	for id := range uniqueMap {
-		unique = append(unique, id)
+	sa, _, err := ssc.setupNewAllocation(
+		request,
+		Timings{timings: nil, start: time.Now()},
+		&transaction.Transaction{
```

Comment:
it's not good, can we use event's db instead?

Reply:
No. The whole point here is to use common code, so when new allocation request is updated then `allocation-min-lock` is updated as well.

History suggests that If we don't do this, then the first time `new allocation request` is modified `allocation-min-lock` will stop working.

Reply:
Changed so we can get the blobbers from the event database. setUpNewAllocation does not use the mpt.